src/createStore.ts

```diff
@@ -39,35 +39,54 @@ import { kindOf } from './utils/kindOf'
  * @returns A Redux store that lets you read the state, dispatch actions
  * and subscribe to changes.
  */
+
+// 3 overloaded functions
+// 1st overload
+//     v--'defualt' allows createStore function to be called by different name when imported later 
 export default function createStore<
-  S,
-  A extends Action,
-  Ext = {},
-  StateExt = never
+  S, // generic of type PreloadedState found in the sotre.ts file and defined below (let currentState = preloadedState as S)
+  A extends Action, // generic that extends the action interface found in the action.ts file
+  Ext = {}, // generic store extension with a default value of an empty object
+  StateExt = never // generic state extension with a default value of never
 >(
-  reducer: Reducer<S, A>,
-  enhancer?: StoreEnhancer<Ext, StateExt>
-): Store<ExtendState<S, StateExt>, A, StateExt, Ext> & Ext
+  reducer: Reducer<S, A>, // 1st (required) param of type Reducer, in all 3 overloads (see notes above from Redux authors)
+  enhancer?: StoreEnhancer<Ext, StateExt> // 1st (optional) param of type StoreEnhancer, in all 3 overloads (see notes above from Redux authors)
+): Store< // return type of Store, which is composed of the joining of ExtendedState & Ext types 
+    ExtendState<
+            S, 
+            StateExt
+        >, 
+        A, 
+        StateExt, 
+        Ext
+        > & // ampersand joins ExtendedState & Ext types
```

Comment:
Never knew about using `&` in a type position for defining an intersection type. Very cool!

Reply:
Yeah. I totally had to look that one up! I have never seen it in JS either.